### DISCUSSION TITLE
Sniffnet is one year old today: lessons learned and next steps

### DISCUSSION THREAD

**Discussion by GyulyVGC** [GyulyVGC/sniffnet#329] [Announcements]
I’m excited to share with y’all that Sniffnet is **one year old today**!
The last 365 days of my life have been almost totally dedicated to this project and I’ve learned a lot along the way.

On August 1st 2022, I’d have never imagined this project would have become what’s today.
Sniffnet was originally started in the scope of an **academic course** and **went much farther** than that: I fell in love with the project and countless additional features and improvements came to my mind.

It’s an unspeakable good feeling to have made a software people use and appreciate, giving something back to this awesome community.
Sometimes it still feels unreal that Sniffnet is now one of the most popular network analysers on GitHub, having passed **10k stars** just a few days ago and being in the [top 100 most starred Rust repositories](https://github.com/EvanLi/Github-Ranking/blob/master/Top100/Rust.md) ever made.

<p align="center">
<img alt="" src="https://github.com/GyulyVGC/sniffnet/assets/100347457/02abb972-255e-4fa2-ab4e-9d02512f9d19"  />
</p>

<p>
<a href="#x">
<img alt="" src="https://github.com/GyulyVGC/sniffnet/blob/main/resources/repository/hr.png?raw=true" width="100%"/>
</a>
</p>

### **Lessons learned**

I cannot say to have drastically improved my coding skills, but what I can assert with certainty is to have understood how **developing software isn’t just programming**: it requires providing an adequate documentation, user support, packaging, distribution, and opens the possibility for exchange of ideas with other folks, that is one of the most interesting and stimulating parts.

I had the one-in-a-life opportunity to be part of the first [GitHub Accelerator](https://accelerator.github.com) cohort, which not only gave me important insights about open-source but also allowed me to work **full-time** on Sniffnet during the past months.

 I’d also like to mention the little yet **heart-warming individual donations** I’ve received on PayPal, Patreon, and GitHub Sponsors: generous people do really exist, often they’re hidden just out there (a big shoutout to @0x0177b11f, @Ilmi2, @Cthulu201, @ZEROF, @wahn).

<p>
<a href="#x">
<img alt="" src="https://github.com/GyulyVGC/sniffnet/blob/main/resources/repository/hr.png?raw=true" width="100%"/>
</a>
</p>

### Next steps

After the sponsorship by GitHub, the financial support has not been enough to permit me keep working on this project full-time.
It’s unfortunate not because of the stream of money itself, but because I truly love developing Sniffnet, and I’d like to be able to spend more and more time on it.
Despite this, I can guarantee that **Sniffnet will be constantly updated and maintained in the future**, even if not at the pace of the past months.

Some of the features planned for the **future** are:
- PIDs identification
- Read and write of PCAP files
- Alerts based on the IP addresses reputation
- Details about unassigned IPs
- New filtering capabilities
- Support for ICMP
- More details about each notification
- Additional settings and configurations

<p>
<a href="#x">
<img alt="" src="https://github.com/GyulyVGC/sniffnet/blob/main/resources/repository/hr.png?raw=true" width="100%"/>
</a>
</p>

### Wrap up

This project has **changed my life forever**, no matter if I won't work on it full-time for the rest of my days.

I was suggested by some friends to make Sniffnet a **freeware**, introducing features reserved to paid users, and I was really tempted to do it with the functionalities introduced in [v1.2](https://github.com/GyulyVGC/sniffnet/releases/tag/v1.2.0).
On the other hand, the open-source ecosystem gave me a lot and it didn’t feel right to turn away.
Eventually, **I decided to listen none other than my heart**: Sniffnet is and will remain forever and ever **completely open-source** software.

**Passion is the engine that pushes us forward, that fills us with motivation, satisfaction, enthusiasm, and hope.
Listen to your heart, do what you enjoy, and enjoy what you do: this is an outstanding method to spend life to the fullest.**

**Comment by Murffy99** [comment]
Deep

**Comment by orhun** [comment]
Great write-up @GyulyVGC 👏🏼 I can definitely feel the passion behind it.

As a close follower of your journey, I can say that you've accomplished so many things and I am looking forward to see what future will bring for you/Sniffnet 🚀 

Good luck with everything! 🐻 

\- friendly neighborhood Arch maintainer

**Reply by GyulyVGC** [reply]
Thank you Orhun 🙏

You are a true example for many people in this ecosystem, me included, and receiving support from you is something unique 🙌

**Comment by borisorlov21** [comment]
Отлично у тебя получилось
